docs/topics/tour/kotlin-tour-intermediate-classes-interfaces.md

```diff
@@ -269,87 +269,185 @@ For more information about interfaces and interface inheritance, see [Interfaces
+If you want to create a class similar to the `PenTool` class which has the same behavior **except** for the value of the
+`color` property, you can. First, create a new class that expects an object that implements the `DrawingTool` interface as a parameter,
+like an instance of the `PenTool` class. Then, inside the class, override the `color` property.
+
+But in this scenario, you need to add implementations for each member of the `DrawingTool` interface:
 
 ```kotlin
-class RedCircle(val circle: Circle) : Circle {
+interface DrawingTool {
+    val color: String
+    fun draw(shape: String)
+    fun erase(area: String)
+    fun getToolInfo(): String
+}
+
+class PenTool : DrawingTool {
+    override val color: String = "black"
+
+    override fun draw(shape: String) {
+        println("Drawing $shape using a pen in $color")
+    }
+
+    override fun erase(area: String) {
+        println("Erasing $area with pen tool")
+    }
+
+    override fun getToolInfo(): String {
+        return "PenTool(color=$color)"
+    }
+}
+//sampleStart
+class CanvasSession(val tool: DrawingTool) : DrawingTool {
+    override val color: String = "blue"
 
-    // Start of boilerplate code
-    override fun draw() {
-        circle.draw()
+    override fun draw(shape: String) {
+        tool.draw(shape)
     }
 
-    override fun resize() {
-        circle.resize()
+    override fun erase(area: String) {
+        tool.erase(area)
     }
 
-    // End of boilerplate code
-    override val color = "red"
+    override fun getToolInfo(): String {
+        return tool.getToolInfo()
+    }
+}
+//sampleEnd
+fun main() {
+    val pen = PenTool()
+    val session = CanvasSession(pen)
+
+    println("Pen color: ${pen.color}")
+    // Pen color: black
+
+    println("Session color: ${session.color}")
+    // Session color: blue
+
+    session.draw("circle")
+    // Drawing circle with pen in black
+
+    session.erase("top-left corner")
+    // Erasing top-left corner with pen tool
+
+    println(session.getToolInfo())
+    // PenTool(color=black)
 }
 ```
+{kotlin-runnable="true" id="kotlin-tour-interface-non-delegation"}
 
-You can see that if you have a large number of member functions in the `Drawable` interface, the amount of boilerplate
-code in the `RedCircle` class can be very large. However, there is an alternative.
+You can see that if you have a large number of member functions in the `DrawingTool` interface, the amount of boilerplate
+code in the `CanvasSession` class can be very large. However, there is an alternative.
 
-In Kotlin, you can use delegation to delegate the interface implementation to an instance of a class. For example,
-you can create an instance of the `Circle` class and delegate the implementations of the member functions of the `Circle`
-class to this instance. To do this, use the `by` keyword. For example:
+In Kotlin, you can use delegation to delegate the interface implementation to an instance of a class. To do this, use 
```

Comment:
IMO, too verbose.

In Kotlin, you can delegate the interface implementation to another class instance using the `by` keyword.

Will it work?